code/modules/power/prototype_shield_emitter.dm

```diff
@@ -0,0 +1,208 @@
+
+/obj/machinery/power/proto_sh_emitter/anchored
+	anchored = TRUE
+
+/obj/machinery/power/proto_sh_emitter/small
+	name = "Small Prototype Shield Emitter"
+	desc = "This is the less powerful version of the P.S.E. designed to be used in the incinerator"
+	circuit = /obj/item/circuitboard/machine/proto_sh_emitter_small
+
+/obj/machinery/power/proto_sh_emitter/small/anchored
+	anchored = TRUE
+
+/obj/machinery/power/proto_sh_emitter/Destroy()
+	if(SSticker.IsRoundInProgress())
+		var/turf/T = get_turf(src)
+		message_admins("Prototype Shield Emitter deleted at [ADMIN_VERBOSEJMP(T)]")
+		log_game("Prototype Shield Emitter deleted at [AREACOORD(T)]")
+	for(var/H in signs)
+		qdel(H)
+	return ..()
+
+/obj/machinery/power/proto_sh_emitter/update_icon_state()
+	if(is_on == TRUE)
+		icon_state = "proto_sh_emitter_on"
+	else
+		icon_state = "proto_sh_emitter"
+
+/obj/machinery/power/proto_sh_emitter/proc/can_be_rotated(mob/user,rotation_type)
+	if(anchored)
+		to_chat(user, "<span class='warning'>It is fastened to the floor!</span>")
+		return FALSE
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/wrench_act(mob/living/user, obj/item/I)
+	if(is_on == TRUE)
+		to_chat(user, "<span class='warning'>You have to turn the [src] off first!</span>")
+		return TRUE
+	anchored = !anchored
+	if(anchored)
+		to_chat(user, "<span class='warning'>You bolt the [src] to the floor!</span>")
+	else
+		to_chat(user, "<span class='warning'>You unbolt the [src] from the floor!</span>")
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/attackby(obj/item/I, mob/user, params)
+	if(I.GetID())
+		if(allowed(user))
+			if(is_on)
+				locked = !locked
+				to_chat(user, "<span class='notice'>You [locked ? "lock" : "unlock"] the controls.</span>")
+			else
+				to_chat(user, "<span class='warning'>The controls can only be locked when \the [src] is online!</span>")
+		else
+			to_chat(user, "<span class='danger'>Access denied.</span>")
+		return
+
+/obj/machinery/power/proto_sh_emitter/crowbar_act(mob/living/user, obj/item/I)
+	default_deconstruction_crowbar(I)
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/screwdriver_act(mob/living/user, obj/item/I)
+	if(..())
+		return TRUE
+	default_deconstruction_screwdriver(user, "proto_sh_emitter-open", "proto_sh_emitter", I)
+	return TRUE
+
+/obj/machinery/power/proto_sh_emitter/process()
+	var/area/a = get_area(src)
+	var/turf/Turf = get_turf(src)
+	if(a.power_equip == FALSE && is_on == TRUE)
+		is_on = FALSE
+		update_icon_state()
+		message_admins("[src] turned off at [ADMIN_VERBOSEJMP(Turf)]")
+		log_game("[src] turned off at [AREACOORD(Turf)]")
+
+/obj/machinery/power/proto_sh_emitter/interact(mob/user)
+	var/area/a = get_area(src)
+	add_fingerprint(user)
+	if(!anchored)
+		to_chat(user, "<span class='warning'>You need to anchor the [src] first!</span>")
+		return
+	if(a.power_equip == FALSE)
```

Comment:
You should be using powered() generally instead of checking area directly.

Reply:
is just if(powered()) ?

Reply:
yes